pinot-broker/src/main/java/org/apache/pinot/broker/routing/timeboundary/TimeBoundaryManager.java

```diff
@@ -65,6 +66,7 @@ public class TimeBoundaryManager {
   private final long _timeOffsetMs;
   private final Map<String, Long> _endTimeMsMap = new HashMap<>();
 
+  private long _explicitlySetTimeBoundaryMs = INVALID_TIME_MS;
```

Comment:
Should be marked @volatile

Reply:
Doesn't need to be because it is always under the synchronized block. Maybe worth adding a comment

Reply:
init() method isn't synchronised though?